torch/csrc/jit/passes/frozen_concat_linear.cpp

```diff
@@ -0,0 +1,255 @@
+#include <c10/util/irange.h>
+#include <torch/csrc/jit/ir/alias_analysis.h>
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/ir/ir_views.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/frozen_concat_linear.h>
+#include <torch/csrc/jit/passes/frozen_conv_folding.h>
+#include <torch/csrc/jit/passes/frozen_graph_optimizations.h>
+#include <torch/csrc/jit/passes/remove_dropout.h>
+#include <torch/csrc/jit/passes/utils/optimization_utils.h>
+#include <torch/csrc/jit/runtime/graph_executor.h>
+#include <torch/csrc/utils/memory.h>
+#include <vector>
+
+namespace torch {
+namespace jit {
+namespace {
+
+using Tensor = at::Tensor;
+
+class ConcatLinearLayers {
+  std::shared_ptr<Graph> graph_;
+  bool graph_modified = false;
+  std::unique_ptr<AliasDb> aliasDb_ = nullptr;
+
+  AliasDb* getAliasDb() {
+    if (!aliasDb_) {
+      aliasDb_ = std::make_unique<AliasDb>(graph_);
+    }
+    return aliasDb_.get();
+  }
+
+  void collectConstantLinearLayers(
+      Block* b,
+      std::unordered_map<Value*, std::list<Node*>>& grouped_linear_layers,
+      std::vector<Value*>& ordered_tensor_inputs) {
+    // We are using an ordered list so that one knows that
```

Comment:
Using a list because it is linear time to remove from the middle of a list. I don't do any indexing into the data structure. (but I am not opposed to using a vector.

Reply:
yea, i comment on that elsewhere. i think removing and iterating through the list is harder/more complicated to follow and slower than the alternative